From d36373f102aa79a618ea058f5f920829ec748d4b Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Fri, 22 Mar 2019 15:05:14 -0700
Subject: [PATCH 0578/1852] v4l: xilinx: vip: Add scaling and padding factor
 for XY10

commit 2176243f4c295fc43f5571a41ea3d22385c28f8c from
https://github.com/Xilinx/linux-xlnx.git

This patch adds scaling and padding factors for stride calculation
of XY10 pixel format.

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vip.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-vip.c b/drivers/media/platform/xilinx/xilinx-vip.c
index c80e2ec59cf9..15a9942811d6 100644
--- a/drivers/media/platform/xilinx/xilinx-vip.c
+++ b/drivers/media/platform/xilinx/xilinx-vip.c
@@ -171,6 +171,7 @@ EXPORT_SYMBOL_GPL(xvip_get_format_by_fourcc);
 void xvip_bpl_scaling_factor(u32 fourcc, u32 *numerator, u32 *denominator)
 {
 	switch (fourcc) {
+	case V4L2_PIX_FMT_XY10:
 	case V4L2_PIX_FMT_XV15:
 	case V4L2_PIX_FMT_XV20:
 	case V4L2_PIX_FMT_XV15M:
@@ -206,6 +207,7 @@ EXPORT_SYMBOL_GPL(xvip_bpl_scaling_factor);
 void xvip_width_padding_factor(u32 fourcc, u32 *numerator, u32 *denominator)
 {
 	switch (fourcc) {
+	case V4L2_PIX_FMT_XY10:
 	case V4L2_PIX_FMT_XV15:
 	case V4L2_PIX_FMT_XV20:
 	case V4L2_PIX_FMT_XV15M:
-- 
2.31.1

